File: hw8/GroupSubmission/process.cpp
/*
    Authors: Dustin Knecht & River Bieberich
    Favorite Location: Trine University Library "The LINK" (1st Floor, Corner by the Stairs)
*/

#include <iostream>
#include <string>
#include <ctime>

#define MAGENTA "\033[35m"
#define RESET   "\033[0m"

std::string timestamp() {
    time_t now = time(NULL);
    char buf[32];
    strftime(buf, sizeof(buf), "%H:%M:%S", localtime(&now));
    return buf;
}

int main(int argc, char* argv[]) {
    if (argc < 2) {
        std::cerr << MAGENTA "[" << timestamp() << "] C++: Missing argument" << RESET << std::endl;
        return 1;
    }

    int value = std::atoi(argv[1]);
    value++;

    std::cout << value << std::endl;

    std::cerr << MAGENTA "[" << timestamp() << "] C++ processed: " << value << RESET << std::endl;

    return 0;
}

File: hw8/GroupSubmission/process.py
# Authors: Dustin Knecht & River Bieberich
# Favorite Location: Trine University Library "The LINK" (1st Floor, Corner by the Stairs)

import sys
import time

CYAN = "\033[36m"
RESET = "\033[0m"

def timestamp():
    return time.strftime("%H:%M:%S")

if len(sys.argv) < 2:
    print("0")
    print(f"{CYAN}[{timestamp()}] Python: Missing argument{RESET}", file=sys.stderr)
    sys.exit(1)

try:
    value = int(sys.argv[1])
except ValueError:
    print("0")
    print(f"{CYAN}[{timestamp()}] Python: Invalid integer{RESET}", file=sys.stderr)
    sys.exit(1)

value += 1
print(value)

print(f"{CYAN}[{timestamp()}] Python processed: {value}{RESET}", file=sys.stderr)